src/pages/standard-report/HtmlReport.js

```diff
@@ -37,26 +38,67 @@ const Loader = () => (
     </div>
 )
 
-const HtmlReport = ({ html }) =>
-    html ? (
-        <iframe
-            id="html-report-id"
-            srcDoc={wrapHtmlInTemplate(html)}
-            title="html-report-content"
-            width="100%"
-            seamless={true}
-            sandbox="allow-same-origin allow-scripts allow-modals"
-        >
-            <style jsx>{`
-                iframe {
-                    border: none;
-                    flex-grow: 1;
-                }
-            `}</style>
-        </iframe>
-    ) : (
-        <Loader />
-    )
+class HtmlReport extends Component {
+    state = { heigth: 'auto' }
+    iframeHtml = null
+    iframeBody = null
+    heightObserver = null
+
+    onIframeLoad = event => {
+        const iframe = event.target
+
+        this.iframeHtml = iframe.contentWindow.document.documentElement
+        this.iframeBody = iframe.contentWindow.document.body
+        this.heightObserver = new window.ResizeObserver(this.adjustHeight)
+
+        this.heightObserver.observe(this.iframeHtml)
+        this.heightObserver.observe(this.iframeBody)
+        this.adjustHeight()
+    }
+
+    adjustHeight = () => {
+        const height = Math.max(
+            this.iframeBody.clientHeight,
+            this.iframeBody.scrollHeight,
+            this.iframeBody.offsetHeight,
+            this.iframeHtml.clientHeight,
+            this.iframeHtml.scrollHeight,
+            this.iframeHtml.offsetHeight
+        )
+
+        // Add 20px as a "safety margin" in case we get a horizontal scroll bar
+        this.setState({ height: height + 20 })
+    }
+
+    componentWillUnmount() {
+        this.heightObserver && this.heightObserver.unobserve(this.iframeHtml)
+        this.heightObserver && this.heightObserver.unobserve(this.iframeBody)
+    }
+
+    render() {
+        return this.props.html ? (
+            <iframe
+                id="html-report-id"
+                srcDoc={wrapHtmlInTemplate(this.props.html)}
+                title="html-report-content"
+                width="100%"
+                height={this.state.height}
+                seamless={true}
+                sandbox="allow-same-origin allow-scripts allow-modals"
+                onLoad={this.onIframeLoad}
```

Comment:
I experimented with several other approaches to initialise the height observers:
- A `ref` and the `componentDidMount` method
- A function `ref`
The problem with both these methods is that the `iframe` itself is available at this point, but it's content hasn't loaded yet. Interestingly, at this point I would be able to access the `html` and `body` element of the iframe, but these would be different elements than the ones available once the iframe's `onLoad` callback ran. I'm not quite sure why, I suppose the iframe is rendered first with some default content and then gets populated via the `srcDoc` attribute. So attaching the `ResizeObserver` to the `html` and `body` elements at this point was useless as they would soon be removed and the resize callback would never run.

`onLoad` runs after the iframe has been populated via the `srcDoc` so at this point we can set references to the iframe's html and body that remain valid until the component unmounts.